backup/moodle2/backup_hvp_stepslib.php

```diff
@@ -15,68 +15,107 @@
-        /*
+
         // To know if we are including userinfo
         $userinfo = $this->get_setting_value('userinfo');
 
         // Define each element separated
-        $choice = new backup_nested_element('choice', array('id'), array(
-            'name', 'intro', 'introformat', 'publish',
-            'showresults', 'display', 'allowupdate', 'showunanswered',
-            'limitanswers', 'timeopen', 'timeclose', 'timemodified',
-            'completionsubmit'));
-
-        $options = new backup_nested_element('options');
-
-        $option = new backup_nested_element('option', array('id'), array(
-            'text', 'maxanswers', 'timemodified'));
-
-        $answers = new backup_nested_element('answers');
-
-        $answer = new backup_nested_element('answer', array('id'), array(
-            'userid', 'optionid', 'timemodified'));
+        $hvp = new backup_nested_element('hvp', array('id'), array(
+            'course',
+            'name',
+            'json_content',
+            'embed_type',
+            'disable',
+            'main_library_id',
+            'content_type',
+            'author',
+            'license',
+            'meta_keywords',
+            'meta_description',
+            'filtered',
+            'slug',
+            'timecreated',
+            'timemodified',
+        ));
+
+            $hvp_libraries = new backup_nested_element('libraries');
+                $hvp_library = new backup_nested_element('library', null, array(
+                    'machine_name',
+                    'major_version',
+                    'minor_version',
+                    'patch_version',
+                    'dependency_type',
+                    'drop_css',
+                    'weight',
+                ));
+
+            $content_user_data_entries = new backup_nested_element('content_user_data');
+                $content_user_data = new backup_nested_element('entry', array('id'), array(
+                    'user_id',
+                    'hvp_id',
+                    'sub_content_id',
+                    'data_id',
+                    'data',
+                    'preloaded',
+                    'delete_on_content_change',
+                ));
 
         // Build the tree
-        $choice->add_child($options);
-        $options->add_child($option);
+        $hvp->add_child($content_user_data_entries);
+            $content_user_data_entries->add_child($content_user_data);
 
-        $choice->add_child($answers);
-        $answers->add_child($answer);
+        $hvp->add_child($hvp_libraries);
+            $hvp_libraries->add_child($hvp_library);
 
         // Define sources
-        $choice->set_source_table('choice', array('id' => backup::VAR_ACTIVITYID));
-
-        $option->set_source_table('choice_options', array('choiceid' => backup::VAR_PARENTID), 'id ASC');
+        $hvp->set_source_table('hvp', array('id' => backup::VAR_ACTIVITYID));
+
+            $lib_sql = "SELECT
```

Comment:
The indentations are deliberate. They mimic the way how the elements are nested within the backup structure.

But it's true that this isn't part of Moodle's coding conventions. I can align them if you want.